### PR TITLE
Fix v8 environment variable buffer overrun 

### PR DESCRIPTION
**Pull request by Dronplane**
### Scope & Purpose

Fixes reading environment variable for windows.
First attempt is made via static 8k buffer. And only if it is not enough dynamic allocation is made.

- [x] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [x] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.10: *(Please link PR)*
  - [ ] Backport for 3.9: *(Please link PR)*
  - [ ] Backport for 3.8: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: tests/test-definitions.txt [hunk 1]
```diff
@@ -87,7 +87,7 @@ recovery_cluster priority=2000 cluster buckets=8 -- --test search
 shell_server_aql priority=1000 cluster buckets=16 -- --dumpAgencyOnError true
 shell_client priority=500 cluster buckets=5 -- --dumpAgencyOnError true
 shell_client_transaction priority=500 cluster parallelity=5 buckets=5 -- --dumpAgencyOnError true
-#shell_client_replication2_recovery priority=500 cluster buckets=5 -- --dumpAgencyOnError true
+shell_client_replication2_recovery priority=500 cluster buckets=5 -- --dumpAgencyOnError true
```
Comment (jsteemann) [inline tests/test-definitions.txt:90]: This is a potentially unrelated change. Is it intentional?

Comment (Dronplane) [reply tests/test-definitions.txt:90]: Yes, this test was blocked by this issue. At least @dothebart  asked me to uncomment it.

**Comment by Dronplane** [comment]
https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/25967/

File: tests/test-definitions.txt (hunk 1 above)
Comment (jsteemann) [reply tests/test-definitions.txt:90]: Thanks for clarifying!

**Review by jsteemann — APPROVED** [review]
LGTM

**Review by dothebart — APPROVED** [review]

**Review by KVS85 — APPROVED** [review]
